src/DependencyInjection/Compiler/TweakCompilerPass.php

```diff
@@ -114,7 +114,7 @@ private function replaceBlockName(ContainerBuilder $container, Definition $defin
         $arguments = $definition->getArguments();
 
         // Replace empty block id with service id
-        if (empty($arguments) || 0 === \strlen($arguments[0])) {
+        if (empty($arguments) || null === ($arguments[0]) || \is_string($arguments[0]) && 0 === \strlen($arguments[0])) {
```

Comment:
Maybe you should introduce a special method for this check

Reply:
If you want, but I'm not really convinced… there is more distance now, and the comment already clarified what this was for.